### PR TITLE
[2.4] Enhance launcher executor

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
Fixes 4561872

### Description
- Rename "external_execution_wait" to "external_pre_init_timeout"
- Enhance logging

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by yanchengnv** [comment]
Where are the changes to set timeout for notify_abort and notify_end?

**Comment by YuanTingHsieh** [comment]
@yanchengnv they are in this PR: https://github.com/NVIDIA/NVFlare/pull/2413

File: nvflare/app_common/executors/launcher_executor.py [hunk 1]
```diff
@@ -277,36 +277,40 @@ def _execute_launcher_method_in_thread_executor(self, method_name: str, **kwargs
...
 
     def _finalize_external_execution(
         self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort_signal: Signal
     ) -> bool:
-        with self._lock:
-            if self._job_end:
-                ask_peer_end_success = self.ask_peer_to_end(fl_ctx)
-                if not ask_peer_end_success:
-                    return False
+        if self._job_end:
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/app_common/executors/launcher_executor.py:303]: we don't need that here, we are not setting stuff here

File: nvflare/app_common/executors/launcher_executor.py [hunk 2]
```diff
@@ -63,18 +63,19 @@ def __init__(
...
-            monitor_interval (float): Interval for monitoring the launcher (default: 0.01).
-            read_interval (float): Interval for reading from the pipe (default: 0.5).
-            heartbeat_interval (float): Interval for sending heartbeat to the peer (default: 5.0).
-            heartbeat_timeout (float): Timeout for waiting for a heartbeat from the peer (default: 30.0).
-            workers (int): Number of worker threads needed (default: 1).
-            train_with_evaluation (bool): Whether to run training with global model evaluation (default: True).
-            train_task_name (str): Task name of train mode (default: train).
-            evaluate_task_name (str): Task name of evaluate mode (default: evaluate).
-            submit_model_task_name (str): Task name of submit_model mode (default: submit_model).
+            external_setup_timeout (float): Time to wait for external process to start.
```
Comment (YuanTingHsieh) [inline nvflare/app_common/executors/launcher_executor.py:68]: Change to this name:

external_pre_init_timeout (float): Time to wait for external process before it calls flare.init().

**Pull request merged**